src/colorizer/CanvasOverlay.ts

```diff
@@ -90,47 +191,202 @@ export default class CanvasOverlay {
+   * @returns an object with two properties:
+   *  - `size`: a vector representing the width and height of the rendered scale bar, in pixels.
+   *  - `render`: a callback that renders the scale bar to the canvas.
+   */
+  private renderTimestamp(ctx: CanvasRenderingContext2D, originPx: Vector2): RenderInfo {
+    if (!this.timestampOptions.visible) {
+      return { sizePx: new Vector2(0, 0), render: () => {} };
+    }
+
+    ////////////////// Format timestamp as text //////////////////
+    const timestampFormatted = CanvasOverlay.getTimestampLabel(this.timestampOptions);
+
+    // TODO: Would be nice to configure top/bottom/left/right padding separately.
+    const timestampPaddingPx = new Vector2(6, 2);
+    const timestampOriginPx = new Vector2(originPx.x + timestampPaddingPx.x, originPx.y + timestampPaddingPx.y);
+    // Save the render function for later.
+    const render = (): void => {
+      this.renderRightAlignedText(ctx, timestampOriginPx, timestampFormatted, this.scaleBarOptions);
+    };
+
+    return {
+      sizePx: new Vector2(
+        timestampPaddingPx.x * 2 + this.getTextDimensions(ctx, timestampFormatted, this.timestampOptions).x,
+        timestampPaddingPx.y * 2 + this.timestampOptions.fontSizePx
+      ),
+      render,
+    };
+  }
+
+  /**
+   * Draws the background overlay in the bottom right corner of the canvas.
+   * @param ctx Canvas context to render to.
+   * @param size Size of the background overlay.
+   * @param options Configuration for the background overlay.
+   */
+  private static renderBackground(ctx: CanvasRenderingContext2D, size: Vector2, options: BackgroundOptions): void {
+    ctx.fillStyle = options.fill;
+    ctx.strokeStyle = options.stroke;
+    ctx.beginPath();
+    ctx.roundRect(
+      Math.round(ctx.canvas.width - size.x - options.marginPx.x) + 0.5,
+      Math.round(ctx.canvas.height - size.y - options.marginPx.y) + 0.5,
+      Math.round(size.x),
+      Math.round(size.y),
+      options.radiusPx
+    );
+    ctx.fill();
+    ctx.stroke();
+    ctx.closePath();
+  }
+
+  /**
+   * Render the overlay to the canvas.
+   */
   render(): void {
     const ctx = this.canvas.getContext("2d");
     if (ctx === null) {
       console.error("Could not get canvas context");
       return;
     }
+
     //Clear canvas
     ctx.clearRect(0, 0, this.canvas.width, this.canvas.height);
 
-    // Draw scale bar and other elements
-    this.renderScaleBar();
+    // Get dimensions + render methods for the elements, but don't render yet so we can draw the background
+    // behind them.
+    const origin = this.backgroundOptions.marginPx.clone().add(this.backgroundOptions.paddingPx);
+    const { sizePx: scaleBarDimensions, render: renderScaleBar } = this.getScaleBarRenderer(ctx, origin);
+    origin.y += scaleBarDimensions.y;
+    const { sizePx: timestampDimensions, render: renderTimestamp } = this.renderTimestamp(ctx, origin);
```

Comment:
consider naming `getScaleBarRenderer` and `renderTimestamp` in a more consistent way, since they seem to return the same type of thing.